### ISSUE TITLE
Evaluate diary-style expression entries

### ISSUE THREAD

**Issue by przemekd** [nvim-orgmode/orgmode#195]
### Does this feature exist in Emacs orgmode core?

Yes

### Orgmode link

https://orgmode.org/manual/Weekly_002fdaily-agenda.html

### Feature value

By using diary-style expression entries such as:
```
%%(diary-remind '(diary-anniversary 06 02 1969) 14) %d. wedding anniversary
```
one can easily add anniversaries and holidays into the agenda.
Right now it seems to be unsupported.

**Comment by kristijanhusak** [comment]
@przemekd did you use the diary style within emacs? I never used it, and syntax seems a bit confusing. If you have experience with it, providing brief explanation would be super helpful.

**Comment by przemekd** [comment]
@kristijanhusak yeah, I do use it. I admit, it is confusing but also really useful at the same time. I don't think I can explain it better than the official documentation: https://www.gnu.org/software/emacs/manual/html_node/emacs/Special-Diary-Entries.html

`diary-remind` used in the example above creates a reminder that a particular date comes close.

**Comment by TravonteD** [comment]
As a note for this `%%()` is just calling a function `diary-remind`, so I think generic support for calling functions using the `%%()` syntax would get us most of the way, then the function diary-remind could be implemented in this repo (or in a plugin)

**Comment by przemekd** [comment]
@TravonteD Do you mean adding support for calling functions [here](https://github.com/milisims/tree-sitter-org) first? Are there any materials on how to start developing something like that? I could try to help here and try to implement it myself.
